File: Python/I_Palindrome.py
def main():
    num = int(input())
    reversed_num = reverseNum(num)
    print(reversed_num)
    print("YES" if num == reversed_num else "NO")


def reverseNum(num):
    ans = 0
    while num:
        last_digit = num % 10
        ans = ans * 10 + last_digit
        num //= 10
    return ans


if __name__ == "__main__":
    main()

File: C/I_Palindrome.c
#include <stdio.h>


int reverseNum(int num)
{
    int ans = 0;
    while (num)
    {
        int lastDigit = num % 10;
        ans = ans * 10 + lastDigit;
        num /= 10;
    }
    return ans;
}


int main() 
{
    int num;
    scanf("%d", &num);

    int reversedNum = reverseNum(num);
    printf("%d\n", reversedNum);
    if (num == reversedNum) printf("YES\n");
    else printf("NO\n");

    return 0;
}